contingency/contingency-api/src/test/java/com/powsybl/contingency/SubContingenciesProviderTest.java

```diff
@@ -47,15 +45,30 @@ public void test() {
 
     @Test
     public void testEmpty() {
-        ContingenciesProvider provider = n -> Collections.emptyList();
+        ContingenciesProvider provider = ContingenciesProviders.emptyProvider();
 
         Network network = Mockito.mock(Network.class);
 
-        List<String> subList1 = new SubContingenciesProvider(provider, new Partition(1, 1))
+        List<String> subList1 = ContingenciesProviders.newSubProvider(provider, new Partition(1, 1))
                 .getContingencies(network)
                 .stream().map(Contingency::getId).collect(Collectors.toList());
 
         assertEquals(Collections.emptyList(), subList1);
     }
 
+    private static ContingenciesProvider newContingenciesProvider() {
+        return new ContingenciesProvider() {
+            @Override
+            public List<Contingency> getContingencies(Network network) {
+                return IntStream.range(1, 5)
+                        .mapToObj(i -> Contingency.builder("contingency-" + i).build())
+                        .collect(Collectors.toList());
+            }
+
+            @Override
+            public String asScript() {
+                return "";
+            }
+        };
+    }
```

Comment:
I don't understand why you change this test. None of these changes have been introduced in #925

Reply:
Bad commit